common/src/main/java/com/wynntils/features/user/AbbreviateMobHealthFeature.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.config.Config;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.mc.event.BossHealthUpdateEvent;
+import com.wynntils.mc.mixin.accessors.ClientboundBossEventPacketAccessor;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.network.protocol.game.ClientboundBossEventPacket;
+import net.minecraft.network.protocol.game.ClientboundBossEventPacket.AddOperation;
+import net.minecraft.network.protocol.game.ClientboundBossEventPacket.Operation;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class AbbreviateMobHealthFeature extends UserFeature {
+
+    private static final Pattern MOB_HEALTH_PATTERN = Pattern.compile("(.*§c)(\\d+)(§4❤.*)");
+
+    @Config
+    public static int precision = 1;
+
+    @SubscribeEvent
+    public void onHealthBarEvent(BossHealthUpdateEvent event) {
+        ClientboundBossEventPacket packet = event.getPacket();
+        Operation operation = ((ClientboundBossEventPacketAccessor) packet).getOperation();
```

Comment:
My gut reaction is that this should be done by method calls on the BossHealthUpdateEvent, like having a `setName` method. But I realize that the BossHealthUpdatePacket is kind of special and that this really don't make sense. 

Maybe what we really should do is to have a "boss health manager" that help us make this kind of manipulations more easily. I'm not saying you need to fix that in this PR, just throwing out the idea.

Reply:
yeah, the way boss packets work made it somewhat awkward to do this - i'm not sure how many other times it's useful to manipulate boss bar content though

Reply:
I think we have a lot of boss bar interfacing already. And I promise you, if we make it easy to manipulate the boss bar, that we will find tons of new excuses to do it in many more cases. :-)

Reply:
that's fair. i would rather leave that framework for a later PR however